### PR TITLE
Add support for timescales with time-taken access log token.

### PR DESCRIPTION
**Pull request by ChristopherSchultz**
Also reintroduce support for fractional-seconds and add nanosecond timescale.

### CODE REVIEW

File: java/org/apache/catalina/valves/AbstractAccessLogValve.java [hunk 1]
```diff
@@ -1756,10 +1776,14 @@ protected AccessLogElement createAccessLogElement(String name, char pattern) {
                 return new DateAndTimeElement(name);
             case 'T':
                 // ms for milliseconds, us for microseconds, and s for seconds
-                if ("ms".equals(name)) {
-                    return new ElapsedTimeElement(false, true);
+                if ("ns".equals(name)) {
+                    return new ElapsedTimeElement(ElapsedTimeElement.Style.MILLISECONDS);
```
Comment (markt-asf) [inline java/org/apache/catalina/valves/AbstractAccessLogValve.java:1780]: I think you want `NANOSECONDS` here.

File: java/org/apache/catalina/valves/AbstractAccessLogValve.java [hunk 2]
```diff
@@ -1756,10 +1776,14 @@ protected AccessLogElement createAccessLogElement(String name, char pattern) {
...
                 } else if ("us".equals(name)) {
-                    return new ElapsedTimeElement(true, false);
+                    return new ElapsedTimeElement(ElapsedTimeElement.Style.MICROSECONDS);
+                } else if ("ms".equals(name)) {
+                    return new ElapsedTimeElement(ElapsedTimeElement.Style.MILLISECONDS);
+                } else if ("fs".equals(name)) {
+                    return new ElapsedTimeElement(ElapsedTimeElement.Style.SECONDS_FRACTIONAL);
```
Comment (markt-asf) [inline java/org/apache/catalina/valves/AbstractAccessLogValve.java:1786]: Strictly, that is fs is femto seconds which doesn't look right. Maybe sf?

**Review by markt-asf — CHANGES_REQUESTED** [review]

File: java/org/apache/catalina/valves/AbstractAccessLogValve.java (hunk 1 above)
Comment (ChristopherSchultz) [reply java/org/apache/catalina/valves/AbstractAccessLogValve.java]: If course I did. Thanks for spotting that.

File: java/org/apache/catalina/valves/AbstractAccessLogValve.java (hunk 2 above)
Comment (ChristopherSchultz) [reply java/org/apache/catalina/valves/AbstractAccessLogValve.java:1786]: Good question. I actually thought of femtoseconds and then thought "who actually knows about those?" and now I guess I have my answer.

We could use `sf`, sure... also `s.` which is a little obscure. There is no reason is has to be two-letters, either. `fracsec`?

Comment (markt-asf) [reply java/org/apache/catalina/valves/AbstractAccessLogValve.java:1786]: It is the electronics engineer in me - capacitors have really small values.

`fracsec` works for me.

Comment (ChristopherSchultz) [reply java/org/apache/catalina/valves/AbstractAccessLogValve.java:1786]: Done/

**Review by markt-asf — APPROVED** [review]

**Pull request merged**